vector/src/main/java/im/vector/riotx/features/roomdirectory/createroom/CreateRoomViewModel.kt

```diff
@@ -39,8 +41,13 @@ class CreateRoomViewModel @AssistedInject constructor(@Assisted initialState: Cr
 
         @JvmStatic
         override fun create(viewModelContext: ViewModelContext, state: CreateRoomViewState): CreateRoomViewModel? {
-            val fragment: CreateRoomFragment = (viewModelContext as FragmentViewModelContext).fragment()
-            return fragment.createRoomViewModelFactory.create(state)
+            val activity: FragmentActivity = (viewModelContext as ActivityViewModelContext).activity()
+
+            return when (activity) {
+                is CreateRoomActivity    -> activity.createRoomViewModelFactory.create(state)
```

Comment:
Two first cases are doing the same thing?

Reply:
cast is not the same